### PR TITLE
Fix linux VM deployment issues

### LINKED ISSUE #2345: Guacamole Linux VM fails install due to empty sources.list.d and missing public keys.
The keys need to be installed before cloud inti configures the sources.

### PR DESCRIPTION
**Pull request by marrobi**
# Resolves #2345

## What is being addressed

- Remove pypi apt repo - not sure why this is there
- Set clean sources.list.d to force
- Move apt public key retrieval to before apt configuration

### CODE REVIEW

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/get_apt_keys.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+# set -o nounset
```
Comment (ross-p-smith) [inline templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/get_apt_keys.sh:5]: Why is this one commented out?

**Review by ross-p-smith — CHANGES_REQUESTED** [review]

File: templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/get_apt_keys.sh (hunk 1 above)
Comment (marrobi) [reply templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/get_apt_keys.sh:5]: Uncommented.

**Comment by marrobi** [comment]
/test-force-approve (passed: https://github.com/microsoft/AzureTRE/actions/runs/2733872158)

**Review by ross-p-smith — APPROVED** [review]

**Review by tamirkamara — APPROVED** [review]

**Comment by marrobi** [comment]
/test-force-approve (passed: https://github.com/microsoft/AzureTRE/actions/runs/2733872158)

**Pull request merged**